homeassistant/components/homepluscontrol/switch.py

```diff
@@ -0,0 +1,216 @@
+      async_update
+      async_added_to_hass
+
+    The SwitchEntity class provides the functionality of a ToggleEntity and additional power
+    consumption methods and state attributes.
+    """
+
+    def __init__(self, coordinator, idx):
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator)
+        self.idx = idx
+
+    @property
+    def name(self):
+        """Name of the device."""
+        return self.coordinator.data[self.idx].name
+
+    @property
+    def unique_id(self):
+        """ID (unique) of the device."""
+        return self.idx
+
+    @property
+    def device_info(self):
+        """Device information."""
+        return {
+            "identifiers": {
+                # Unique identifiers within the domain
+                (DOMAIN, self.unique_id)
+            },
+            "name": self.name,
+            "manufacturer": "Legrand",
+            "model": HW_TYPE.get(self.coordinator.data[self.idx].hw_type, "Unknown"),
+            "sw_version": self.coordinator.data[self.idx].fw,
+        }
+
+    @property
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        if self.coordinator.data[self.idx].device == "plug":
+            return DEVICE_CLASS_OUTLET
+        return DEVICE_CLASS_SWITCH
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Logger of entity."""
+        return logging.getLogger(__name__)
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available.
+
+        This is the case when the coordinator is able to update the data successfully
+        AND the switch entity is reachable.
+
+        This method overrides the one of the CoordinatorEntity
+        """
+        return (
+            self.coordinator.last_update_success
+            and self.coordinator.data[self.idx].reachable
+        )
+
+    @property
+    def is_on(self):
+        """Return entity state."""
+        self.logger.debug(
+            "Status of %s : %s",
+            str(self.name),
+            str(self.coordinator.data[self.idx].status),
+        )
+        return self.coordinator.data[self.idx].status == "on"
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on."""
+        # Do the turning on.
+        await self.coordinator.data[self.idx].turn_on()
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the entity off."""
+        await self.coordinator.data[self.idx].turn_off()
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from hass."""
+        if self.registry_entry is not None:
+            entity_reg = (
+                await self.coordinator.hass.helpers.entity_registry.async_get_registry()
+            )
```

Comment:
Please import and use  the helper to get the registry from the entity registry instead. This is probably not needed at all though.

https://github.com/home-assistant/core/blob/6ad7020f99f82e86698e4ae1d457d4d2a7ac8dd8/homeassistant/helpers/entity_registry.py#L574-L577

Reply:
Yes, I realized that this is not really needed. Thanks.